libs/frontend/feature-build/src/lib/dnd/row-layout.ts

```diff
@@ -38,74 +38,90 @@ export const isRowLayout = (
-}: IHandleHoverProps) => {
+export const handleRowLayoutHover = (props: IHandleHoverProps) => {
+  const {
+    mousePosition,
+    hoverCmpRect,
+    haveSameParent,
+    dragComponentIndex,
+    hoverCmpIndex,
+    dropProps,
+  } = props
+
+  const { left, right, width } = hoverCmpRect
+  const hoverableWidth = Math.min(8, width * 0.15)
+
   const hoverLeft =
-    mousePosition.x < targetMiddle.x &&
-    (!haveSameParent || dragComponentIndex !== targetIndex - 1)
+    mousePosition.x >= left &&
+    mousePosition.x <= left + hoverableWidth &&
+    (!haveSameParent || dragComponentIndex !== hoverCmpIndex - 1)
 
   const hoverRight =
-    mousePosition.x > targetMiddle.x &&
-    (!haveSameParent || dragComponentIndex !== targetIndex + 1)
+    mousePosition.x <= right &&
+    mousePosition.x >= right - hoverableWidth &&
+    (!haveSameParent || dragComponentIndex !== hoverCmpIndex + 1)
 
   dropProps.hoverLeft = hoverLeft
   dropProps.hoverRight = hoverRight
-  dropProps.destinationIndex = targetIndex
 
   // Calculate destination index
   if (hoverLeft) {
-    if (haveSameParent && dragComponentIndex <= targetIndex) {
-      dropProps.destinationIndex = targetIndex - 1
+    dropProps.destinationIndex = hoverCmpIndex
+    if (haveSameParent && dragComponentIndex <= hoverCmpIndex) {
+      dropProps.destinationIndex = hoverCmpIndex - 1
     }
   } else if (hoverRight) {
+    dropProps.destinationIndex = hoverCmpIndex
     if (haveSameParent) {
-      if (dragComponentIndex > targetIndex) {
-        dropProps.destinationIndex = targetIndex + 1
+      if (dragComponentIndex > hoverCmpIndex) {
+        dropProps.destinationIndex = hoverCmpIndex + 1
       }
     } else {
-      dropProps.destinationIndex = targetIndex + 1
+      dropProps.destinationIndex = hoverCmpIndex + 1
     }
   }
 }
 
-export const handleColumnLayoutHover = ({
-  mousePosition,
-  targetMiddle,
-  haveSameParent,
-  dragComponentIndex,
-  targetIndex,
-  dropProps,
-}: IHandleHoverProps) => {
+export const handleColumnLayoutHover = (props: IHandleHoverProps) => {
+  const {
+    mousePosition,
+    hoverCmpRect,
+    haveSameParent,
+    dragComponentIndex,
+    hoverCmpIndex,
+    dropProps,
+  } = props
+
+  const { top, bottom, height } = hoverCmpRect
+  const hoverableHeight = Math.min(8, height * 0.15)
+
   const hoverTop =
-    mousePosition.y < targetMiddle.y &&
-    (!haveSameParent || dragComponentIndex !== targetIndex - 1)
+    mousePosition.y >= top &&
+    mousePosition.y <= top + hoverableHeight &&
+    (!haveSameParent || dragComponentIndex !== hoverCmpIndex - 1)
 
   const hoverBottom =
-    mousePosition.y > targetMiddle.y &&
-    (!haveSameParent || dragComponentIndex !== targetIndex + 1)
+    mousePosition.y <= bottom &&
+    mousePosition.y >= bottom - hoverableHeight &&
+    (!haveSameParent || dragComponentIndex !== hoverCmpIndex + 1)
 
   dropProps.hoverTop = hoverTop
   dropProps.hoverBottom = hoverBottom
-  dropProps.destinationIndex = targetIndex
```

Comment:
This strange adjustment is intended to fix a new bug where the component will be moved up one level even if it's not being dropped on the top side of its parent.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/4bdf764a-d213-4ea4-9435-e2a6c736b6bf